homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -246,7 +294,10 @@ def __init__(
         self._observations = [
             Observation(
                 entity_id=observation.get(CONF_ENTITY_ID),
-                platform=observation[CONF_PLATFORM],
+                # YAML uses the key CONF_PLATFORM but ConfigFlow uses CONF_TYPE
```

Comment:
Sorry, my mistake but let's stick with the same terminology so use `platform` instead of `type`.

Reply:
Done in https://github.com/home-assistant/core/pull/122552/commits/a33b075ad873e2d01678ee02c59be9dd5cb7cd3d